File: week-2/isomorphic-strings/isomorphic-strings.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isIsomorphic = function(s, t) {
    if (s.length !== t.length) return false;
    return createMapping(s) === createMapping(t);
};

const createMapping = (string) => {
    const hashtable = {};
    let counter = 0
    let mapping = '';
    string.split('').forEach(i => {
        if (!hashtable.hasOwnProperty(i)) {
            hashtable[i] = counter;
            ++counter;
        }
        mapping += hashtable[i];
    });
    
    return mapping;
};

File: week-2/isomorphic-strings/isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        if len(s) != len(t): return False
        
        return self.createMapping(s) == self.createMapping(t)
        
    def createMapping(self, string):
        mapping = ''
        counter = 0
        hashtable = {}
        for i in string:
            if i not in hashtable:
                hashtable[i] = counter
                counter += 1
            mapping += str(hashtable[i])
            
        return mapping
